lambda-buffers-compiler/src/LambdaBuffers/Compiler/TypeClass/Pat.hs

```diff
@@ -19,10 +19,14 @@ cost of significantly more complex type signatures.
 -}
 
 data Pat
-  = {- extremely stupid, unfortunately necessary -}
+  = {- Name / ModuleName / Opaque / TyVarP are literal patterns (or ground terms)
+       because hey cannot contain any VarPs and therefore "have no holes".
+       Every TyDef or subcomponent thereof will be translated into a composite
+       pattern "without any holes". (Nil is also a literal/ground term, I guess) -}
```

Comment:
They're literal patterns. 

```
--- yes i know this is a stupid way to write this 
isDrazen :: String -> Bool 
isDrazen str = case str of
  "Drazen" -> True     -- "Drazen" is a literal pattern here. It's not in an expression position so it can't be a (literal) expression 
   _ -> False          -- '_' is a wildcard pattern, which we don't have b/c they're not useful for us  
```

In our context they only match w/ identical literal patterns (in Haskell they match the identical literal expression). 

As an aside, it would be more "proper" to have an `Expr` type which is identical to `Pat` but w/o `VarP`. It would be _ideal_ to just use the ProtoCompat types, but we can't do that (without doing something ridiculous) because the proto AST isn't defined recursively (so there's no simple way to write `match`). 

I think a lot of your confusion/discomfort here comes from the fact that I used `Pat` to represent both proper patterns and "expressions" (patterns w/o `VarP` holes). If you think it would help I can go back and actually make an `Expr` type. (In the Haskell AST you can turn a pattern w/o `VarP` or `WildP` patterns into an expression simply by changing all of the LitP/ListP/etc values into LitE/ListE/etc values, so what I didn't isn't that wild really)